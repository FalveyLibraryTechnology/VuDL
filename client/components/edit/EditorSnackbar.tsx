import React from "react";
import Alert from "@mui/material/Alert";
import IconButton from "@mui/material/IconButton";
import Snackbar from "@mui/material/Snackbar";
import CloseIcon from "@mui/icons-material/Close";
import { useEditorContext } from "../../context/EditorContext";

const EditorSnackbar = (): React.ReactElement => {
    const {
        state: {
            snackbarState: { message, open, severity },
        },
        action: { setSnackbarState },
    } = useEditorContext();

    const handleClose = () => {
        setSnackbarState({
            open: false,
            message: "",
            severity: "info",
        });
    };

    return (
        <Snackbar className="editorSnackbar" open={open} autoHideDuration={5000} onClose={handleClose}>
            <Alert
                className="editorSnackbarAlert"
                severity={severity}
                action={
                    <IconButton
                        className="editorSnackBarAlertCloseButton"
                        aria-label="close"
                        color="inherit"
                        size="small"
                        onClick={handleClose}
                    >
                        <CloseIcon fontSize="inherit" />
                    </IconButton>
                }
            >
                {message}
            </Alert>
        </Snackbar>
    );
};

export default EditorSnackbar;
