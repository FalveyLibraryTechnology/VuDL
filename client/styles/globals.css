:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
