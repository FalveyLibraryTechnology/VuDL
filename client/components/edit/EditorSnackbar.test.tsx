import React from "react";
import { beforeEach, describe, expect, it, jest } from "@jest/globals";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import EditorSnackbar from "./EditorSnackbar";

const mockUseEditorContext = jest.fn();
jest.mock("../../context/EditorContext", () => ({
    useEditorContext: () => {
        return mockUseEditorContext();
    },
}));
jest.mock("./ChildList", () => () => "ChildList");

describe("EditorSnackbar", () => {
    let editorValues;
    beforeEach(() => {
        editorValues = {
            state: {
                snackbarState: {
                    message: "test1",
                    open: true,
                    severity: "info",
                },
            },
            action: {
                setSnackbarState: jest.fn(),
            },
        };
        mockUseEditorContext.mockReturnValue(editorValues);
    });

    it("renders", () => {
        const wrapper = shallow(<EditorSnackbar />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("closes", () => {
        const component = mount(<EditorSnackbar />);

        component.find("button.editorSnackBarAlertCloseButton").simulate("click");

        expect(editorValues.action.setSnackbarState).toHaveBeenCalledWith({
            open: false,
            message: "",
            severity: "info",
        });
    });
});
