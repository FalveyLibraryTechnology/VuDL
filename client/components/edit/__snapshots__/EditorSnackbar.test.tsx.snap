// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorSnackbar renders 1`] = `
<div
  className="MuiSnackbar-root MuiSnackbar-anchorOriginBottomLeft editorSnackbar css-cpgvjg-MuiSnackbar-root"
  onBlur={[Function]}
  onClick={[Function]}
  onClose={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchEnd={[Function]}
  role="presentation"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard editorSnackbarAlert css-1fuz9qr-MuiPaper-root-MuiAlert-root"
    direction="up"
    role="alert"
    style={
      {
        "opacity": 1,
        "transform": "scale(1, 1)",
        "visibility": undefined,
      }
    }
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="InfoOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      test1
    </div>
    <div
      className="MuiAlert-action css-ki1hdl-MuiAlert-action"
    >
      <button
        aria-label="close"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall editorSnackBarAlertCloseButton css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
